### PR TITLE
FS.COM (Fiberstore) 'GBN' and 'SWITCH' devices support

### PR DESCRIPTION
**Pull request by PipoCanaja**
Following https://github.com/librenms/librenms/issues/9712 and https://github.com/librenms/librenms/issues/9117

Mibs are quite buggy but some data is polled (Serial, HW, SW Version, CPU and Memory) on some devices. 

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Cormoran96** [comment]
Works fine for me. Thanks

**Comment by PipoCanaja** [comment]
@Cormoran96 
Which type of device ? Which sensors are discovered ? 
Thx

**Comment by Cormoran96** [comment]
My bad.

I thought the processor graph start graph but no. 

screenshot from 2019-01-28 09-55-50

**Comment by PipoCanaja** [comment]
So the 3900 family is different than the 3800 ... Their SNMP implementation is really bad ... 

If you can collect snmpwalk as described here, I'll check if something can be done : 
https://docs.librenms.org/Support/FAQ/#faq20

**Comment by Cormoran96** [comment]
> If you can collect snmpwalk as described here, I'll check if something can be done 

I already do that here  #9712

File: html/images/os/fs.svg [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (murrant) [inline html/images/os/fs.svg]: This needs to be optimized, I usually just paste it into https://jakearchibald.github.io/svgomg/

**Comment by murrant** [comment]
Looks good overall.  I'm a but unsure about the OS name.
According to the MIBs, it should probably be called GBN.
The enterprise ID in the mib is GCOM Technologies, these are either rebranded GCOM or that is just a random enterprise ID:

```
--      Note that the admin enterprise ID (13464) is used in the sample
--      MIBs.  The customer "must" replace the admin enterprise ID with
--      the registered enterprise identifier for the specific company or
--      product.  The customer must also replace the ADMIN information
--      located in the 'MODULE-IDENTITY' in each sample MIB.
```

**Comment by PipoCanaja** [comment]
In fact, it seems that they use multiple different mibs, from different origins ... One of them is indeed GBN according to the mib filenames. 
But as only very few OIDs are available, I suggest to stick with one OS (and handle the different OIDs in it) instead of creating as many OSes as devices from this vendor...

File: LibreNMS/OS/Fs.php [hunk 2]
```diff
@@ -55,6 +55,8 @@ public function discoverProcessors()
         }
         // Tests OID from SWITCH MIB.
         $processors_data = snmpwalk_cache_oid($this->getDevice(), 'ssCpuIdle', [], 'SWITCH', 'fs');
+
+        $processors = [];
```
Comment (PipoCanaja) [inline LibreNMS/OS/Fs.php]: Not good. If you do so, you potentially forget the processors that were discovered in the previous foreach. 
New commit on its way.

**Comment by PipoCanaja** [comment]
@murrant whenever you have time to review this, I would suggest we merge it. If new models of this vendor are to be integrated, we'll extend this later.
From my point of view, it looks OK.

File: includes/definitions/fs.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+over:
+over:
+    - { graph: device_bits, text: 'Device Traffic' }
+    - { graph: device_processor, text: 'Processor Usage' }
+    - { graph: device_mempool, text: 'Memory Usage' }
+mib_dir:
+    - fs
+discovery:
+    - sysObjectID: 
+        - .1.3.6.1.4.1.51134
```
Comment (murrant) [inline includes/definitions/fs.yaml]: That is a lot of sysObjectIDs.  I'm concerned this OS isn't the right scope.  OS should be defined per OS/MIB set, not per vendor.

Comment (PipoCanaja) [reply includes/definitions/fs.yaml]: OK. I don't really understand the mess they did with so many sysObjectIDs. For the moment we have 2 mib groups (GBN* and SWITCH) so we will start with 2 OSes grouped in "fs".

Comment (murrant) [reply includes/definitions/fs.yaml]: I'm pretty sure all of the Fiberstore equipment is just rebranded oem equipment.

Comment (PipoCanaja) [reply includes/definitions/fs.yaml]: That would make sense ... Anyway, I'll still do that split ASAP.

Comment (PipoCanaja) [reply includes/definitions/fs.yaml]: Split done. Right now, 2 groups.

File: includes/discovery/mempools/fs-switch.inc.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+// Discovery for FS-switch devices.
+//
+
+if ($device['os'] == 'fs-switch') {
+    echo 'FS : ';
+    $mempools_array = snmpwalk_cache_oid($device, 'memTotalReal', [], 'SWITCH', 'fs');
+    $mempools_array = snmpwalk_cache_oid($device, 'memTotalFree', $mempools_array, 'SWITCH', 'fs');
+    $mempools_array = snmpwalk_cache_oid($device, 'memTotalUsed', $mempools_array, 'SWITCH', 'fs');
+    d_echo($mempools_array);
+    if (is_array($mempools_array)) {
```
Comment (murrant) [inline includes/discovery/mempools/fs-switch.inc.php]: is_array is always true, you can just remove the if

**Comment by murrant** [comment]
Looking pretty close.  Just one thing I commented on.  Also curious as to why you picked `fs-other`.

File: includes/discovery/mempools/fs-switch.inc.php (hunk 4 above)
Comment (PipoCanaja) [reply includes/discovery/mempools/fs-switch.inc.php]: done

**Comment by PipoCanaja** [comment]
I ended up with fs-gbn and fs-switch according to the mib they follow. You should not find any fs-other anymore in the current status of the PR.

**Pull request merged**